cmd/team-updater/main.go

```diff
@@ -0,0 +1,169 @@
+//
+// Add or remove users from PTO:
+//   (pto|nopto) users...
+//
+// Usernames are matched first on exact Github username, and secondly on the
+// user's real name.
+//
+// Examples:
+//   backporter aanm joe  # set backports to André and Joe
+//   pto jarno            # mark Jarno as being on PTO
+//   nopto jussi          # mark Jussi as being back from PTO
+package main
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/cilium/team-manager/pkg/config"
+	"github.com/cilium/team-manager/pkg/persistence"
+	"gopkg.in/yaml.v2"
+
+	flag "github.com/spf13/pflag"
+)
+
+// A stringSet is a set of strings.
+type stringSet map[string]struct{}
+
+var (
+	configFilename string
+	dryRun         bool
+)
+
+func init() {
+	flag.StringVar(&configFilename, "config-filename", "cilium-team-assignments.yaml", "Config filename")
+	flag.BoolVar(&dryRun, "dry-run", false, "Dry run the steps without performing any write operation to GitHub")
+	flag.Parse()
+}
+
+func findUser(config *config.Config, s string) (string, error) {
+	// First, try to find users by exact match of the Github username.
+	if _, ok := config.Members[s]; ok {
+		return s, nil
+	}
+
+	// Second, try to find githubUsernames by substring matching their name.
+	var githubUsernames []string
+	for githubUsername, user := range config.Members {
+		if strings.Contains(strings.ToLower(user.Name), strings.ToLower(s)) {
+			githubUsernames = append(githubUsernames, githubUsername)
+		}
+	}
+	switch len(githubUsernames) {
+	case 0:
+		return "", fmt.Errorf("%s: user not found", s)
+	case 1:
+		return githubUsernames[0], nil
+	default:
+		return "", fmt.Errorf("%s: ambiguous user (found %s)", s, strings.Join(githubUsernames, ", "))
+	}
+}
+
+func run() error {
+	c, err := persistence.LoadState(configFilename)
+	if err != nil {
+		return err
+	}
+
+	if flag.NArg() < 2 {
+		return fmt.Errorf("syntax: %s backporter|janitor|triage|pto|nopto users...", os.Args[0])
+	}
+
+	var users []string
+	for i := 1; i < flag.NArg(); i++ {
+		user, err := findUser(c, flag.Arg(i))
+		if err != nil {
+			return err
+		}
+		users = append(users, user)
+	}
+
+	switch flag.Arg(0) {
+	case "nopto":
+		excludeCRAFromAllTeams := newStringSet(c.ExcludeCRAFromAllTeams...)
+		excludeCRAFromAllTeams.remove(users...)
+		c.ExcludeCRAFromAllTeams = excludeCRAFromAllTeams.elements()
+	case "pto":
+		excludeCRAFromAllTeams := newStringSet(c.ExcludeCRAFromAllTeams...)
+		excludeCRAFromAllTeams.add(users...)
+		c.ExcludeCRAFromAllTeams = excludeCRAFromAllTeams.elements()
+	default:
+		team := flag.Arg(0)
+		teamConfig, ok := c.Teams[team]
+		if !ok {
+			return fmt.Errorf("%s: unknown team", team)
+		}
+		teamConfig.Members = newStringSet(users...).elements()
+		c.Teams[team] = teamConfig
+	}
+
+	if err := config.SanityCheck(c); err != nil {
+		return err
+	}
+
+	if dryRun {
+		return yaml.NewEncoder(os.Stdout).Encode(c)
+	}
+
+	return persistence.StoreState(configFilename, c)
+}
+
+func main() {
```

Comment:
Why not make this as an option for the team-management instead of a new binary?

Reply:
OK, done. It makes the command a little more verbose to use.